### PR TITLE
[0.65] Fix Google Test Adapter detection in CI/PR so unit tests are actually run

### LINKED ISSUE #9191: Native tests are not running in 0.65-stable
The CI for 0.65 broke by a change the agent pool (PR #9105) which was automerged in by the bot despite not passing the checks.

In trying to get that working again, I've discovered that tests weren't always running to begin with.

The Native tests were silently failing even before that pool change, in CI and PR ([example](https://dev.azure.com/ms/react-native-windows/_build/results?buildId=245058&view=logs&j=f0c4580e-283f-520d-cb83-83d60c6c4549&t=334c7169-f3c6-524c-e730-2cab64c4c561)):

```
[MSTest][Discovery][D:\a\1\s\vnext\target\X86\Debug\Microsoft.ReactNative.Cxx.UnitTests\Microsoft.ReactNative.Cxx.UnitTests.exe] Failed to discover tests from assembly D:\a\1\s\vnext\target\X86\Debug\Microsoft.ReactNative.Cxx.UnitTests\Microsoft.ReactNative.Cxx.UnitTests.exe. Reason:Could not load file or assembly 'Microsoft.ReactNative.Cxx.UnitTests' or one of its dependencies. The system cannot find the file specified.
[MSTest][Discovery][D:\a\1\s\vnext\target\X86\Debug\Microsoft.ReactNative.IntegrationTests\Microsoft.ReactNative.IntegrationTests.exe] Failed to discover tests from assembly D:\a\1\s\vnext\target\X86\Debug\Microsoft.ReactNative.IntegrationTests\Microsoft.ReactNative.IntegrationTests.exe. Reason:Could not load file or assembly 'Microsoft.ReactNative.IntegrationTests' or one of its dependencies. The system cannot find the file specified.
[MSTest][Discovery][D:\a\1\s\vnext\target\X86\Debug\Mso.UnitTests\Mso.UnitTests.exe] Failed to discover tests from assembly D:\a\1\s\vnext\target\X86\Debug\Mso.UnitTests\Mso.UnitTests.exe. Reason:Could not load file or assembly 'Mso.UnitTests' or one of its dependencies. The system cannot find the file specified.
No test is available in D:\a\1\s\vnext\target\X86\Debug\Microsoft.ReactNative.Cxx.UnitTests\Microsoft.ReactNative.Cxx.UnitTests.exe D:\a\1\s\vnext\target\X86\Debug\Microsoft.ReactNative.IntegrationTests\Microsoft.ReactNative.IntegrationTests.exe D:\a\1\s\vnext\target\X86\Debug\Mso.UnitTests\Mso.UnitTests.exe. Make sure that test discoverer & executors are registered and platform & framework version settings are appropriate and try again.
```

### PR DESCRIPTION
**Pull request by jonthysell**
Native tests aren't being run in 0.65 stable during CI. The test runner sees the test DLLs, but doesn't see any tests in them, so passes with 0 tests run.

This PR backports the template changes in 0.66 used to detect the Google Test Adapter to 0.65, and also sets the flag for the test task to fail if 0 tests are detected.

This PR disables the specific failing Desktop tests (covered by #8000 and #9397) and the specific Universal test (#9398).

Closes #9191

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/9396)

### CODE REVIEW

**Review by dannyvv — APPROVED** [review]

**Comment by jonthysell** [comment]
Currently blocked because #9398 causes the universal build to fail.

File: .ado/templates/discover-google-test-adapter.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+steps:
```
Comment (JunielKatarn) [inline .ado/templates/discover-google-test-adapter.yml:1]: It seems like overkill to create a full YAML template only to run a few-lines PowerShell script.

Wouldn't it be better to use an actual PowerShell script and invoke it in a task/step?

That would also make it parameterizable, so it works for locating other support DLLs in the VS installation and not only the GTest adapter.

Comment (jonthysell) [reply .ado/templates/discover-google-test-adapter.yml:1]: I'm just backporting what we're using in 0.66, 0.67, and main, not creating a whole new solution.

**Comment by jonthysell** [comment]
I'm able to get the test tasks passing by only disabling the specific failing tests. For universal this is covered by #9398 (which is fixed in 66) and for desktop this is covered by #8000 and #9397 (both of which are still a problem).

**Comment by jonthysell** [comment]
Attempting to backport #7984 to fix the universal test.

**Comment by jonthysell** [comment]
It looks like backporting #7984 fixes #9398.

**Review by NickGerleman — APPROVED** [review]

**Pull request merged**